src/transformers/generation/candidate_generator.py

```diff
@@ -157,6 +157,13 @@ def __init__(
         self.generation_config.return_dict_in_generate = True
         self.generation_config.output_scores = True
 
+        # avoid unnecessary warnings that min_length is larger than max_new_tokens
+        input_length = input_ids.shape[-1]
+        min_new_tokens = self.generation_config.min_new_tokens if self.generation_config.min_new_tokens else 0
```

Comment:
Are we guaranteed the `self.generation_config` has this attribute?

Reply:
Yes, the GenerationConfig usually initiates those to 0 or None, if not indicated by the user. So, we check if it is not None for `min_new_tokens` and then set `min_length` to maximum (code says min, i'll fix it) between user-defined value or the default 0.

Reply:
OK, and can we ever expect to have whisper generate configs being used here, or does the model always just use it's own custom generation code?